Sources/Basic/SyncronizedQueue.swift

```diff
@@ -0,0 +1,58 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import class Foundation.Condition
+
+/// This class can be used as a shared queue between multiple threads providing
+/// thread safe APIs. The elements are stored as optionals so nil can be used as
+/// a sentinel.
```

Comment:
The class doesn't use nil as a sentinel, the client _can_ use it as sentinel. I documented it because its might seem weird to client that why does the class methods enqueue and dequeue take and give optionals.